src/rendezvous/MSC3906Rendezvous.ts

```diff
@@ -92,9 +127,16 @@ export class MSC3906Rendezvous {
             return;
         }
 
-        this._code = JSON.stringify(await this.channel.generateCode(this.ourIntent));
+        const raw = this.v1FallbackEnabled
+            ? await this.channel.generateCode(this.ourIntent)
+            : await this.channel.generateCode(this.ourIntent, this.flow);
+        this._code = JSON.stringify(raw);
     }
 
+    /**
+     *
+     * @returns the checksum of the secure channel if the rendezvous set up was successful, otherwise undefined
+     */
     public async startAfterShowingCode(): Promise<string | undefined> {
```

Comment:
Particularly given this is a public function, please could it have a better doc-comment? What does it actually do, beyond returning a success flag?